siriuspy/siriuspy/clientconfigdb/types/as_diagnostics.py

```diff
@@ -1435,232 +1252,22 @@ def get_dict():
-    [':TRIGGER_SYSID9TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID10RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID10RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID10TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID10TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID11RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID11RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID11TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID11TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID12RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID12RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID12TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID12TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID13RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID13RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID13TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID13TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID14RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID14RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID14TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID14TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID15RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID15RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID15TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID15TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID16RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID16RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID16TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID16TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID17RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID17RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID17TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID17TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID18RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID18RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID18TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID18TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID19RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID19RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID19TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID19TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID20RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID20RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID20TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID20TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID21RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID21RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID21TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID21TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID22RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID22RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID22TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID22TrnSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID23RcvInSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID23RcvSrc-Sel', 0, 0.0],
-    [':TRIGGER_SYSID23TrnOutSel-SP', 0, 0.0],
-    [':TRIGGER_SYSID23TrnSrc-Sel', 0, 0.0],
-    ]
+_fofb_propts = list()
+for phy_trig in range(8):
+    _fofb_propts.extend([
+        [f':TRIGGER{phy_trig}Dir-Sel', 0, 0.0],
+        [f':TRIGGER{phy_trig}DirPol-Sel', 0, 0.0],
+        [f':TRIGGER{phy_trig}RcvLen-SP', 0, 0.0],
+        [f':TRIGGER{phy_trig}TrnLen-SP', 0, 0.0],
+    ])
+for acq_core in ['_LAMP', '_SYSID']:
+    for log_trig in range(24):
```

Comment:
And since we are looping through known quantities, perhaps add constants for the amount of physical triggers and the amount of logical ones

Reply:
done!